src/services/AccommodationService.js

```diff
@@ -0,0 +1,40 @@
+import { accommodations, rooms, accommodationImages } from '../database/models';
+import CommonQueries from './CommonQueries';
+
+/**
+ * @exports
+ * @class AccommodationService
+ */
+class AccommodationService {
+  /**
+   * Travel Admin can be able to create accomodation
+   * @static
+   * @param {object} req  request object
+   * @memberof AccommodationService
+   * @returns {object} data
+   */
+  static async createAccomodation(req) {
+    const newAccommodationObject = {
+      name: req.body.name,
+      cityId: req.body.cityId,
+      address: req.body.address,
+      description: req.body.description,
+      googleCoordinates: req.body.googleCoordinates,
+    };
+    const newAccommodation = await CommonQueries.create(accommodations, newAccommodationObject);
+    Array.from(new Set(req.body.imageUrls)).map(async (imageUrl) => {
+      await CommonQueries.create(accommodationImages,
+        { accommodationId: newAccommodation.id, imageUrl });
+    });
+    const { rooms: userRooms } = req.body;
+    const uniqueRooms = Array.from(new Set(userRooms.map(a => a.id)))
+      .map(id => userRooms.find(a => a.id === id));
+    uniqueRooms.map(async (room) => {
+      room.accommodationId = newAccommodation.id;
+      await CommonQueries.create(rooms, room);
```

Comment:
I believe the maintainability or refactoring of the code will be hard. I recommend chaining arrays at most 2 times. Basically, what you can do is to save arrays in variables then combine them later.